src/libraries/System.Formats.Cbor/tests/Writer/CborWriterTests.Tag.cs

```diff
@@ -158,7 +158,7 @@ public static void WriteInteger_SingleValue_HappyPath(string valueString, string
         [InlineData("1", "c4820001")]
         [InlineData("-1", "c4820020")]
         [InlineData("1.1", "c482200b")]
-        [InlineData("1.000", "c482221903e8")]
```

Comment:
I'm not familiar with cbor. Do we know why this changed in size?

Reply:
Nor am I. I just figured that we are parsing correctly now and returning `1`. So I looked at the test case 2 lines above for `1` and copied that. I can go back and see what value we're returning without the bug fix, but I'm pretty sure it is wrong somehow and this PR is fixing it.

Reply:
It might be good to get confirmation.

The number of hex digits implies it was over the length 32-bits, but it also looks like there is some general encoding (compression?) here so it's unclear whether the change is actually correct (I would guess it is though).

Reply:
@eiriktsarpalis for thoughts

Reply:
The original CBOR encodes `1000e-3`, while the new one is `1e0`, so I presume this is in line with the intended change in the parser.